aptos-move/aptos-vm/src/move_vm_ext/respawned_session.rs

```diff
@@ -0,0 +1,122 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    aptos_vm_impl::AptosVMImpl,
+    data_cache::{AsMoveResolver, StorageAdapter},
+    move_vm_ext::{SessionExt, SessionId},
+};
+use anyhow::{bail, Result};
+use aptos_aggregator::transaction::ChangeSetExt;
+use aptos_gas::ChangeSetConfigs;
+use aptos_state_view::{StateView, StateViewId, TStateView};
+use aptos_types::{
+    state_store::{
+        state_key::StateKey, state_storage_usage::StateStorageUsage, state_value::StateValue,
+    },
+    write_set::{TransactionWrite, WriteSet},
+};
+use move_core_types::vm_status::{err_msg, StatusCode, VMStatus};
+
+#[ouroboros::self_referencing]
+pub struct RespawnedSession<'r, 'l> {
```

Comment:
Can we have documentation please? I find it really hard to work with VM sometimes because things are never documented and connected in a  very unintuitive way😂